api/scpca_portal/test/management/commands/test_create_portal_metadata.py

```diff
@@ -54,31 +52,50 @@ def load_test_data(self):
         self.assertEqual(Sample.objects.all().count(), SAMPLES_COUNT)
         self.assertEqual(Library.objects.all().count(), LIBRARIES_COUNT)
 
-    def test_create_portal_metadata(self):
+    # TODO: After PR #839 is merged into dev, add readme file format testing
+    def assertProjectReadmeContains(self, text, zip_file):
+        self.assertIn(text, zip_file.read(README_FILE).decode("utf-8"))
+
+    @patch("scpca_portal.management.commands.create_portal_metadata.s3.upload_output_file")
+    def test_create_portal_metadata(self, mock_upload_output_file):
+        # Set up the database for test
         self.load_test_data()
-        self.processor.create_portal_metadata(clean_up_output_data=False)
+        # Create the portal metadata computed_file
+        computed_file = self.processor.create_portal_metadata(
+            clean_up_output_data=False, update_s3=True
+        )
+
+        # Test computed_file
+        if computed_file:
+            expected_size = 8469
+            self.assertEqual(computed_file.size_in_bytes, expected_size)
+            mock_upload_output_file.assert_called_once_with(
+                computed_file.s3_key, computed_file.s3_bucket
+            )
+        else:
+            self.fail("No computed file")
```

Comment:
I think here we should be calling the same assertions independent of the command running successfully. That is what we are here to test / ensure.

I rewrote the test below but I think for better coverage we should also assert that all of the 
`GENERATED_PORTAL_METADATA_DOWNLOAD_CONFIG` values have been applied to the computed file as well.

```suggestion
        self.processor.create_portal_metadata(
            clean_up_output_data=False, update_s3=True
        )

        computed_files = ComputedFile.objects.filter(portal_metadata_only=True)
        self.assertEqual(portal_computed_files.count(), 1)
        
        computed_file = portal_computed_files.first()
        self.assertEqual(computed_file.size_in_bytes, 8469)
        mock_upload_output_file.assert_called_once_with(
            computed_file.s3_key, computed_file.s3_bucket
        )
```

Reply:
Thank you for the rewrite, it looks great! I've also applied your suggestion to assert that all field values match the download config file at d1f8a256a9fe6c9edda6e79b438f354a34e174f9 👍!